pkg/kev/services_test.go

```diff
@@ -141,6 +143,190 @@ var _ = Describe("ServiceConfig", func() {
+
+						err := serviceConfig.validate()
+						Expect(err).To(Succeed())
+					})
+				})
+			})
+
+			Context("tcp probe validation", func() {
+				Context("readiness", func() {
+					BeforeEach(func() {
+						serviceConfig.Labels[config.LabelWorkloadLivenessProbeType] = kubernetes.ProbeTypeNone.String()
+						serviceConfig.Labels[config.LabelWorkloadReadinessProbeType] = kubernetes.ProbeTypeTCP.String()
+					})
+
+					It("fails when tcp probe has missing port", func() {
+						err := serviceConfig.validate()
+						Expect(err).To(HaveOccurred())
+						Expect(err.Error()).To(ContainSubstring("required"))
+						Expect(err.Error()).To(ContainSubstring(config.LabelWorkloadReadinessProbeTCPPort))
+					})
+
+					It("succeeds when port is provided", func() {
+						serviceConfig.Labels[config.LabelWorkloadReadinessProbeTCPPort] = "8080"
+
+						err := serviceConfig.validate()
+						Expect(err).To(Succeed())
+					})
+				})
+				Context("liveness", func() {
+					BeforeEach(func() {
+						serviceConfig.Labels[config.LabelWorkloadLivenessProbeType] = kubernetes.ProbeTypeTCP.String()
+					})
+
+					It("fails when tcp probe has missing port", func() {
+						err := serviceConfig.validate()
+						Expect(err).To(HaveOccurred())
+						Expect(err.Error()).To(ContainSubstring("required"))
+						Expect(err.Error()).To(ContainSubstring(config.LabelWorkloadLivenessProbeTCPPort))
+					})
+
+					It("succeeds when port is provided", func() {
+						serviceConfig.Labels[config.LabelWorkloadLivenessProbeTCPPort] = "8080"
+
+						err := serviceConfig.validate()
+						Expect(err).To(Succeed())
+					})
+				})
+			})
+
+			Context("command probe validation", func() {
+				Context("rediness probess", func() {
+					BeforeEach(func() {
+						serviceConfig.Labels[config.LabelWorkloadLivenessProbeType] = kubernetes.ProbeTypeNone.String()
+						serviceConfig.Labels[config.LabelWorkloadReadinessProbeType] = kubernetes.ProbeTypeCommand.String()
+					})
+
+					It("fails when command probe has missing properties", func() {
+						err := serviceConfig.validate()
+						Expect(err).To(HaveOccurred())
+						Expect(err.Error()).To(ContainSubstring("required"))
+						Expect(err.Error()).To(ContainSubstring(config.LabelWorkloadReadinessProbeCommand))
+					})
+				})
+
+				Context("liveness probe", func() {
+					BeforeEach(func() {
+						serviceConfig.Labels[config.LabelWorkloadLivenessProbeType] = kubernetes.ProbeTypeCommand.String()
+					})
+
+					It("fails when command probe has missing properties", func() {
+						err := serviceConfig.validate()
+						Expect(err).To(HaveOccurred())
+						Expect(err.Error()).To(ContainSubstring("required"))
+						Expect(err.Error()).To(ContainSubstring(config.LabelWorkloadLivenessProbeCommand))
+					})
+				})
+			})
+
+			Context("liveness probe type enum validation", func() {
+				BeforeEach(func() {
+					serviceConfig.Labels[config.LabelWorkloadLivenessProbeType] = kubernetes.ProbeTypeNone.String()
```

Comment:
I think it'd be better to move this `BeforeEach` block to relevant context, for clarity, or remove entirely if not used.

Reply:
This should go under readiness